update/updateutil/updateutil.go

```diff
@@ -0,0 +1,24 @@
+package updateutil
+
+import (
+	"errors"
+	"os"
+	"path/filepath"
+
+	"github.com/mattn/go-zglob"
+)
+
+// ExpandGlobPattern expands a glob pattern in the specified repository path.
+// It reproduces similar error characteristics as filepath.Glob().
+// Supports recursive glob patterns using **
+func ExpandGlobPattern(repoPath, filePathPattern string) ([]string, error) {
```

Comment:
I would put that function in `internal/glob` or `internal/files` pattern or something like that. Just to avoid the "xyzUtil" type of packages where you end up putting all kind of stuff.

Reply:
done